goshopify.go

```diff
@@ -53,7 +54,9 @@ type RateLimitInfo struct {
 // Client manages communication with the Shopify API.
 type Client struct {
 	// HTTP client used to communicate with the Shopify API.
-	Client *http.Client
+	Client  *http.Client
+	log     LeveledLoggerInterface
+	logging bool
```

Comment:
The `logging` variable doesn't make sense to me, the name is way too generic (it really just controls if req and resp are logged) and why not use the loglevel instead? (like you had before if i recall right)

Reply:
well no guarantee the logger that matches the interface contains a member called Level which is all i was testing before. I felt this was a better implementation to have the go-shopify keep a flag if the WithLogger option is called so in the library you can do if c.logging {} so you can remove cpu or io intensive logs if you do not want them. open to better ideas

Reply:
follow up.. the other option was a nil logger but you'll get panics everywhere hence why NewClient adds a leveled logger with no Level. felt a simple bool sitting next to the interface would be simpler to check in the long run

Reply:
I still don't think this is a good way to handle the logging exclusion down in e.g. doGetHeadersDebugResponse()

Reply:
I think you can use default noop logger implementation to avoid panics (i.e. when WithLogger is not passed as a parameter, the default implementation will be assigned, which will basically do nothing)

Reply:
There are some benefits in each approach
Having a separate flag to enable / disable logging will allow you do disable logging without removing the logger itself
Having noop logger by default will look much more cleaner and with less chances to make a mistake (i.e. forget to enable / disable logs)
So I am fine with either approach, but I like the idea of noop logger more

Reply:
I think it's inconsistent and confusing to have a flag called `logging` that controls *some* of the log output and then some is controlled by the Logger itself. The implicit contract between logging and a function like `doGetHeadersDebugRequest` bothers me.

Maybe it's a naming issue to make this more clear, renaming `logging` to something that expresses this better (useCustomLogger ? not sure if that's better. Maybe logRequestResponseBodies`? It's long but at least that's what the flag controls  )) 

I'd also see if we can rename `doGetHeadersDebugRequest` (and it's companion) to something more expressive, `doLogRequest|ResponseBody()` or so.

Reply:
I agree with @oliver006 on naming 

@luthermonson, please could you improve naming, or investigate the idea of noop default logger?